### PR TITLE
Feat/eng 488 integration tests (Part 2: Delegation)

### PR DESCRIPTION
**Pull request by gjgd**
- Tests for Delegation: Bonding, Unbonding, Withdrawing tokens

- Add JS Linter + fix JS errors

### CODE REVIEW

File: test/unit/TransmuteDPOS.spec.js [hunk 1]
```diff
@@ -67,7 +65,7 @@ contract('TransmuteDPOS', accounts => {
       assert.equal(42, totalBondedAmount);
     });
 
-    it("should register a Provider's parameters", async () => {
+    it('should register a Provider\'s parameters', async () => {
       assert.equal(PROVIDER_UNREGISTERED, await tdpos.providerStatus.call(accounts[1]));
       await approveBondProvider(10, 20, 2, 35, 1, accounts[1]);
```
Comment (OR13) [inline test/unit/TransmuteDPOS.spec.js]: these numbers: 10, 20, 2, 35, 1, should be constants or variables with descriptive names.

File: test/unit/TransmuteDPOS.spec.js [hunk 2]
```diff
@@ -316,14 +311,13 @@ contract('TransmuteDPOS', accounts => {
         args: {
           _delegator: accounts[4],
           _provider: accounts[0],
-          _amount: 300
-        }
+          _amount: 300,
```
Comment (OR13) [inline test/unit/TransmuteDPOS.spec.js]: 300 should be a constants or variable

File: test/integration/DPOS.spec.js [hunk 3]
```diff
@@ -135,5 +140,116 @@ contract('integration/TransmuteDPOS', accounts => {
...
+
+    it('delegator3 delegates to provider2', async () => {
+      await tdpos.bond(provider2, 100, {from: delegator3});
+    });
+
+    it('delegator4 delegates to himself', async () => {
+      await tdpos.approve(contractAddress, 100, {from: delegator4});
+      await tdpos.bond(delegator4, 100, {from: delegator4});
+    });
+  });
```
[the remark this answers is not in the document]
Comment (gjgd) [reply test/integration/DPOS.spec.js]: This is tested in the unit tests for bond() and unbond() respectively.

[the remark this answers is not in the document]
Comment (gjgd) [reply test/integration/DPOS.spec.js]: Actually it is missing from bond()'s unit tests. Let me add that

**Pull request merged**